ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRuleTest.kt

```diff
@@ -1871,6 +1871,303 @@ internal class WrappingRuleTest {
             .hasLintViolation(2, 10, "A newline was expected before 'Bar'")
             .isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated var initialisation then wrap each expression to a new line`() {
+        val code =
+            """
+            fun f1() {
+                val a = 3; val b = 2
+                val key: String = ""; val lambda: () -> Unit = {  }
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun f1() {
+                val a = 3;
+                val b = 2
+                val key: String = "";
+                val lambda: () -> Unit = {  }
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
```

Comment:
In tests like this also add `.addAdditionalRuleProvider { NoSemicolonsRule() }` so that the formatted result also has cleaned up the unwanted semicolon.

Reply:
But as per the doc of `addAdditionalRuleProvider`, it runs before the rule which is getting unit tested `An additional rule provider for a rule which actually is executed before the rule under test,`

Reply:
It does not always lead to problems. You can always try to add it anyways and document it like below:
```
                .hasLintViolations(
                    // Some violations below are only reported during linting but not on formatting. After wrapping the
                    // nullable type, there is no need to wrap the parameters.
                    LintViolation(2, 12, "Expected new line before function type as it does not fit on a single line"),
                    LintViolation(2, 13, "Parameter should start on a newline"),
                    LintViolation(2, 24, "Parameter should start on a newline"),
                    LintViolation(2, 35, "Parameter should start on a newline"),
                    LintViolation(2, 44, "Missing newline before \")\""),
                    LintViolation(2, 53, "Expected new line after function type as it does not fit on a single line"),
                ).isFormattedAs(formattedCode)
```

Reply:
I have added `addAdditionalRuleProvider` and it really cleaned semicolons 😅 It seems even after `An additional rule provider for a rule which actually is executed before the rule under test,` due to constraints `NoSemicolonsRule` make sure that it is run after the `WrappingRuleTest`

Reply:
Also, I have not added `addAdditionalRuleProvider` as I also wanted to test this rule w/o `NoSemicolonsRule`

Reply:
Downside is that it is now sometime confusing why the `NoSemicolonsRule` is not present in some test while it is added in other tests. As the default behavior of ktlint is to run both rules, I prefer to add the rule because the formattedCode in that case looks closest to the normal behavior of ktlint.
But when there is a specific need in a test to also validate the end result in case the rule is disabled, then I would suggest to make it explicit in the name of the test or with a comment why the the  `NoSemicolonsRule` is not added.

Reply:
Yes this makes more sense. I have added one TC `Given two variables run without NoSemicolonsRule`